### ISSUE TITLE
`cyclic-class-definition` example from the docs does not report the error

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3715]
https://docs.astral.sh/ty/reference/rules/#cyclic-class-definition

`main.py`:

```py
# foo.pyi
class A(B): ... # reports "unresolved-reference" instead
class B(A): ...
```

[Playground](https://play.ty.dev/92b89007-1798-4b1c-bd5e-c398c6ad9950)

**Comment by DetachHead** [comment]
oh nvm it's only reported in pyi files
